packages/gp2-components/src/organisms/UserProjects.tsx

```diff
@@ -0,0 +1,154 @@
+});
+const listElementStyles = css({
+  display: 'flex',
+  alignItems: 'flex-start',
+  gap: '32px',
+  marginBottom: rem(8),
+});
+const listElementMainStyles = css({
+  [`@media (min-width: ${tabletScreen.min}px)`]: {
+    flex: '40% 0 0',
+  },
+});
+const listElementSecondaryStyles = css({
+  [`@media (min-width: ${tabletScreen.min}px)`]: {
+    flex: '30% 0 0',
+  },
+});
+const headingListStyles = css({
+  margin: '0',
+  flex: '30% 0 0',
+  [`@media (min-width: ${tabletScreen.min}px)`]: {
+    display: 'none',
+  },
+});
+const buttonWrapperStyles = css({
+  paddingTop: rem(8),
+  display: 'flex',
+  justifyContent: 'center',
+  width: '100%',
+  borderBottom: `transparent`,
+});
+
+type UserProjectsProps = Pick<
+  gp2.UserResponse,
+  'id' | 'firstName' | 'projects'
+>;
+const UserProjects: React.FC<UserProjectsProps> = ({
+  projects,
+  firstName,
+  id,
+}) => {
+  const minimumProjectsToDisplay = 3;
+  const [expanded, setExpanded] = useState(false);
+
+  const getProjectListStyles = () => {
+    if (projects.length < minimumProjectsToDisplay + 1 || expanded) {
+      return rowStyles;
+    }
+
+    return [rowStyles, hideStyles];
+  };
+
+  const getUserProjectRole = (
+    userId: gp2.UserResponse['id'],
+    project: gp2.UserResponse['projects'][number],
+  ): gp2.ProjectMemberRole | null =>
+    project.members.find((member) => member.userId === userId)?.role || null;
+
+  return (
+    <>
+      <div css={[contentStyles]}>
+        <Paragraph hasMargin={false} accent="lead">
+          {firstName} has been involved in the following GP2 projects:
+        </Paragraph>
+      </div>
+      <div css={headingTopStyles}>
+        <h4 css={headingTopMainStyles}>Name</h4>
+        <h4 css={headinTopSecondaryStyles}>Role</h4>
+        <h4 css={headinTopSecondaryStyles}>Status</h4>
+      </div>
+      {projects.map((project, index) => (
+        <div key={`user-project-${index}`} css={getProjectListStyles()}>
+          <div css={[listElementStyles, listElementMainStyles]}>
+            <h4 css={headingListStyles}>Name:</h4>
+            <Link underlined={true} href="https://google.com">
+              {project.title}
+            </Link>
+          </div>
+          <div css={[listElementStyles, listElementSecondaryStyles]}>
+            <h4 css={headingListStyles}>Role:</h4>
+            <span css={textStyles}>{getUserProjectRole(id, project)}</span>
+          </div>
+          <div css={[listElementStyles, listElementSecondaryStyles]}>
+            <h4 css={headingListStyles}>Status:</h4>
+            <StatusPill status={project.status} />
+          </div>
+        </div>
+      ))}
+      {projects.length > minimumProjectsToDisplay && (
+        <div css={buttonWrapperStyles}>
+          <Button linkStyle onClick={() => setExpanded(!expanded)}>
+            <span
+              css={{
+                display: 'inline-grid',
+                verticalAlign: 'middle',
+                paddingRight: rem(12),
+              }}
```

Comment:
all the other stylings are extracted to a variable (even if they are used once)

Reply:
this is copied from https://github.com/yldio/asap-hub/blob/dace4dcfa6933c8a6b1063d6fbde1efb130b59ed/packages/gp2-components/src/organisms/ProjectMilestones.tsx#L73-L78